docs/environment.md

```diff
@@ -0,0 +1,186 @@
+used in the build, installation & execution process.
+
+The default usage of this script is to be sourced. For example, local
+configuration file may be constructed as:
+
+```bash
+   export PDO_INSTALL_ROOT=${PDO_SOURCE_ROOT}/build/_dev
+   export PDO_STL_KEY_ROOT=${PDO_INSTALL_ROOT}/opt/pdo/etc/keys/sawtooth
+   export PDO_LEDGER_URL=http://127.0.0.1:8008
+```
+and before building it you call the configuration script as
+
+```bash
+   source ${PDO_SOURCE_ROOT}/build/common-config.sh
+```
+
+If passed the parameter `--evalable-export` the script will return a
+list of export commands of the variables instead of directly exporting
+them to the environment.
+
+Passing parameter `--reset-keys` will unset key variables
+`PDO_ENCLAVE_CODE_SIGN_PEM`, `PDO_LEDGER_KEY_SKF`,
+`PDO_SPID` and `PDO_SPID_KEY_CERT_FILE_PEM` before setting variables.
+
+<!-- -------------------------------------------------- -->
+<!-- -------------------------------------------------- -->
+## Generic Environment Variables
+
+<!-- -------------------------------------------------- -->
+### `PDO_INSTALL_ROOT`
+(default: `${PDO_SOURCE_ROOT}/build/_dev`):
+
+`PDO_INSTALL_ROOT` is the root of the directory in which the virtual
+enviroment will be built; generally `PDO_HOME` will point to
+`PDO_INSTALL_ROOT/opt/pdo`
+
+<!-- -------------------------------------------------- -->
+### `PDO_HOME`
+(default: `${PDO_INSTALL_ROOT}/opt/pdo`):
+
+`PDO_HOME` is the directory where PDO-specific files are stored for
+operation. These files include configuration files, data files, compiled
+contracts, contract user keys and service scripts.
+
+<!-- -------------------------------------------------- -->
+### `PDO_DEBUG_BUILD`
+(default: 0)
+
+`PDO_DEBUG_BUILD` builds PDO modules for debugging. This includes
+compile flags, logging statements in the enclave, etc. Since
+`PDO_DEBUG_BUILD` potentially exposes information about what is
+happening inside a contract, do not use with confidential contracts.
+
+<!-- -------------------------------------------------- -->
+### `TINY_SCHEME_SRC`
+(default: `${PDO_SOURCE_ROOT}/tinyscheme-1.41`)
+
+`TINY_SCHEME_SRC` points to the installation of the tinyscheme source in
+order to build the library used to debug and test contracts outside of
+the contract enclave.
+
+<!-- -------------------------------------------------- -->
+<!-- -------------------------------------------------- -->
+## SGX Environment Variables
+
+<!-- -------------------------------------------------- -->
+### `SGX_MODE`
+(default: `SIM`)
+
+`SGX_MODE` determines the SGX mode of operation. When the variable is
+set to `SIM`, then the SGX enclaves will be compiled for simulator
+mode. When the variable is set to `HW`, the enclaves will be compiled to
+run in a real SGX enclave.
+
+<!-- -------------------------------------------------- -->
+### `PDO_SGX_KEY_ROOT`
+(default: `${PDO_SOURCE_ROOT}/build/keys/sgx_simulation/`):
```

Comment:
The default with SGX_MODE=HW is `${PDO_SOURCE_ROOT}/build/keys/sgx_hardware_mode`

Reply:
i think that would confuse more than help.
maybe a better solution would be to change the directory name and then set the default to
${PDO_SOURCE_ROOT}/build/keys/${SGX_MODE}/

Reply:
agreed -- opened issue #213 for this